src/nvim/buffer_defs.h

```diff
@@ -664,12 +664,13 @@ struct file_buffer {
   char_u      *b_p_efm;         /* 'errorformat' local value */
   char_u      *b_p_ep;          /* 'equalprg' local value */
   char_u      *b_p_path;        /* 'path' local value */
-  int b_p_ar;                   /* 'autoread' local value */
+  bool b_p_ar;                  // 'autoread' local value, when present
+  bool has_b_p_ar;              // whether buffer has an 'autoread' local value
```

Comment:
Is there precedent for this for any of the other options? Seems a bit too subtle and will make merges from upstream harder. 

If we want to do a total refactor like @ZyX-I and @tarruda seem to, it would be better to wait for that mega-change than deal with subtle differences in the interim.